### PR TITLE
Corentin/cert fix

### PR DESCRIPTION
**Pull request by Corentin-pro**
Fix #690

If this snippet is OK I will add the same block in the other snippet and a separate function to call in all samples

### CODE REVIEW

**Review by lukekim — APPROVED** [review]

File: snippet.py [hunk 1]
```diff
@@ -1,7 +1,24 @@
+import os
+from pathlib import Path
+import ssl
+import urllib.request
+
 from pyarrow import flight
 
-client = flight.connect('grpc+tls://flight.spiceai.io')
 
+# Check for gRPC required Google certificate file
```
Comment (lukekim) [inline snippet.py]:
```suggestion
# Check for gRPC certificates
```

File: snippet.py [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
+if not Path('/usr', 'share', 'grpc', 'roots.pem').exists():
+    env_name = 'GRPC_DEFAULT_SSL_ROOTS_FILE_PATH'
+    if env_name not in os.environ or not Path(os.environ[env_name]).exists():
+        tls_root_certs = './roots.pem'
+        if not Path(tls_root_certs).exists():
+            print('Downloading gRPS root certificates')
```
Comment (lukekim) [inline snippet.py]:
```suggestion
            print('Downloading gRPC certificates')
```

File: snippet.py [hunk 3]
```diff
@@ -1,7 +1,24 @@
...
+            # Do not verify certs for https (Mac issue)
```
Comment (lukekim) [inline snippet.py]: I'd just remove this comment.

File: snippet.py [hunk 4]
```diff
@@ -1,7 +1,24 @@
...
+from pathlib import Path
+import ssl
+import urllib.request
+
 from pyarrow import flight
 
-client = flight.connect('grpc+tls://flight.spiceai.io')
 
+# Check for gRPC required Google certificate file
+if not Path('/usr', 'share', 'grpc', 'roots.pem').exists():
```
Comment (lukekim) [inline snippet.py]: Should we add an OS check here? E.g. no point doing this on Windows.

**Review by lukekim — CHANGES_REQUESTED** [review]

File: snippet.py (hunk 3 above)
Comment (Corentin-pro) [reply snippet.py]: OK, done.

File: snippet.py (hunk 4 above)
Comment (Corentin-pro) [reply snippet.py]: Windows seems to have the same issue (according to official gRPC package issue) so this should always return True thus checking the environment variable and avoiding the certificate issue.

So in practice this test should work (should we test it?), I checked the gRPC code and the hard-coded path in indeed platform dependent (https://github.com/grpc/grpc/blob/33ec154cee644dc6b3aa8b8d5e0ece683c681edc/include/grpc/impl/codegen/port_platform.h). If I am reading the code write on windows the environment variable must be used (or in-code argument with the path of the certificate.)

**Review by lukekim — APPROVED** [review]

File: snippet.py (hunk 4 above)
Comment (lukekim) [reply snippet.py]: Let's support Windows in our SDK. May want to add a note in docs.

**Pull request merged**